src/SAML2/XML/md/KeyDescriptor.php

```diff
@@ -30,9 +31,9 @@ class KeyDescriptor
     /**
      * The KeyInfo for this key.
      *
-     * @var \SAML2\XML\ds\KeyInfo|null
+     * @var \SAML2\XML\ds\KeyInfo
      */
-    private $KeyInfo = null;
+    private $KeyInfo;
```

Comment:
This one can still be `null`, and should therefore default to that. Even though the standard mandates it as required, we can have a situation where we create a `KeyDescriptor` object without passing any XML to it, and therefore if we call `getKeyInfo()` immediately after that, the property would still be `null`. I don't know if that would trigger an error, since the `getKeyInfo()` method has now a return type specified, and we would be returning `null` instead, but in any case, we should reflect the actual case.

Reply:
That should trigger a TypeError, yes.. And toXML would also fail because of the assert in place

Reply:
The exception due to the assert in `toXML()` would be normal behaviour, but the `TypeError` wouldn't. It's kind of a grey zone, because `null` would only be returned when the API is misused, but still I think we should actually return the value reflecting the object has not been set, instead of raising an error.